fahrschein/src/main/java/org/zalando/fahrschein/ManagedCursorManager.java

```diff
@@ -129,8 +129,12 @@ public void onSuccess(String eventName, Cursor cursor) throws IOException {
             } else if (status == 200) {
                 LOG.warn("Cursor for subscription [{}] to event [{}] in partition [{}] with offset [{}] was already committed", subscriptionId, eventName, cursor.getPartition(), cursor.getOffset());
             } else {
-                throw new IOException(String.format("Unexpected status code [%s] for subscription [%s] to event [%s]", status, subscriptionId, eventName));
+                throw new CursorOffsetCommitException(status, cursor, subscriptionId);
```

Comment:
It actually make sense to try and read nakadi response (and expose it to a user), as the information about problem may be written there.

Reply:
Regarding parsing - I meant to try and extract string representation, not to actually do string parsing aferwards

Reply:
Done. Note that when we catch the `IOProblem` we already have the response body, I've just propagated the root cause. When we throw the exception based on status code, I'm getting the body and adding it to the exception, which also logs it.